ml-algorithms/src/main/java/org/opensearch/ml/engine/algorithms/agent/MLChatAgentRunner.java

```diff
@@ -143,7 +150,14 @@ public void run(MLAgent mlAgent, Map<String, String> params, ActionListener<Obje
                     params.put(CHAT_HISTORY, chatHistoryBuilder.toString());
                 }
 
-                runAgent(mlAgent, params, listener, memory, memory.getConversationId());
+                ActionListener<Object> finalListener = regenerateInteractionId != null ? ActionListener.runBefore(listener, () -> {
+                    memory.getMemoryManager().deleteInteraction(regenerateInteractionId, ActionListener.wrap(deleted -> {}, e -> {
```

Comment:
Can the interaction be deleted as part of agent executor? ChatAgent should not be aware if it is a regenerate request and this would also help add regenerate support to other agent runners in future.

Reply:
Good suggestion, move it to agent executor makes more sense.